docs/conventions/documenting_code.md

```diff
@@ -1,119 +1,210 @@
 
-    ```crystal
-    #     unicorn = Unicorn.new
-    #     unicorn.speak
-    ```
+### Code Examples
 
-* Text blocks, for example to show program output, must be surrounded with triple backticks followed by the "text" keyword.
+Code examples can be placed in Markdown code blocks.
+If no language tag is given, the code block is considered to be Crystal code.
 
-    ``````crystal
-    # ```text
-    # "I'm a unicorn"
-    # ```
-    ``````
+```crystal
+# Example:
+# ```
+# unicorn = Unicorn.new
+# unicorn.horns # => 1
+# ```
+class Unicorn
+end
+```
 
-* To automatically link to other types, enclose them with single backticks.
+To designate a code block as plain text, it must be explicitly tagged.
 
-    ```crystal
-    # the `Legendary` module
-    ```
+```crystal
+# Output:
+# ```plain
+# "I'm a unicorn"
+# ```
+def say
+  puts "I'm a unicorn"
+end
+```
 
-* To automatically link to methods of the currently documented type, use a hash like `#horns` or `#index(char)`, and enclose it with single backticks.
+Other language tags can also be used.
 
-* To automatically link to methods in other types, do `OtherType#method(arg1, arg2)` or just `OtherType#method`, and enclose it with single backticks.
+To show the value of an expression inside code blocks, use `# =>`.
 
-    For example:
+```crystal
+1 + 2             # => 3
+Unicorn.new.speak # => "I'm a unicorn"
+```
 
-    ```crystal
-    # Check the number of horns with `#horns`.
-    # See what a unicorn would say with `Unicorn#speak`.
-    ```
+### Admonitions
 
-* To show the value of an expression inside code blocks, use `# =>`.
+Several admonition keywords are supported to visually highlight problems, notes and/or possible issues.
 
-    ```crystal
-    1 + 2             # => 3
-    Unicorn.new.speak # => "I'm a unicorn"
-    ```
+- `BUG`
+- `DEPRECATED`
+- `FIXME`
+- `NOTE`
+- `OPTIMIZE`
+- `TODO`
 
-* Use `:ditto:` to use the same comment as in the previous declaration.
+Admonition keywords must be the first word in their respective line and must be in all caps. An optional trailing colon is preferred for readability.
 
-    ```crystal
-    # :ditto:
-    def number_of_horns
-      horns
-    end
-    ```
+```crystal
+# Makes the unicorn speak to STDOUT
+#
+# NOTE: Although unicorns don't normally talk, this one is special
+# TODO: Check if unicorn is asleep and raise exception if not able to speak
+# TODO: Create another `speak` method that takes and prints a string
+def speak
+  puts "I'm a unicorn"
+end
 
-* Use `:nodoc:` to hide public declarations from the generated documentation. Private and protected methods are always hidden.
+# Makes the unicorn talk to STDOUT
+#
+# DEPRECATED: Use `speak`
+def talk
+  puts "I'm a unicorn"
+end
+```
 
-    ```crystal
-    class Unicorn
-      # :nodoc:
-      class Helper
-      end
-    end
-    ```
+The compiler implicitly adds some admonitions to doc comments:
+
+* The [`@[Deprecated]`](https://crystal-lang.org/api/Deprecated.html) annotation
```

Comment:
```suggestion
* The [`@[Deprecated]`](https://crystal-lang.org/api/latest/Deprecated.html) annotation
```

Reply:
Most links on other pages don't include `latest` in the path. Do we want to standardize on either form? Is there a specific benefit of using `latest`?

Reply:
One less redirect / request.
The urls without /latest/ were the original location before we split it in versions.